cfme/middleware/__init__.py

```diff
@@ -83,7 +83,8 @@ def get_server_name(path):
 
 class Container(SummaryMixin):
 
-    def add_deployment(self, filename, runtime_name=None, enable_deploy=True, cancel=False):
+    def add_deployment(self, filename, runtime_name=None, enable_deploy=True, cancel=False,
+                       expect_success=True):
```

Comment:
If you are trying to test negative assertion, we use the ```with error.expected``` to detail the error we expect to see instead of what we did see.